### PR TITLE
Problem: changeset verify command return wrong app-hash on old blocks

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- skip the stores who are added after the target version

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
 - [#924](https://github.com/crypto-org-chain/cronos/pull/924) memiavl support `Export` API.
 - [#934](https://github.com/crypto-org-chain/cronos/pull/934) Add pebbledb backend.
 - [#950](https://github.com/crypto-org-chain/cronos/pull/950) Implement memiavl and integrate with state machine.
+- [#]() Fix versiondb verify command on older versions
```
Comment (yihuang) [inline CHANGELOG.md:31]:
```suggestion
- [#985](https://github.com/crypto-org-chain/cronos/pull/985) Fix versiondb verify command on older versions
```

Comment (yihuang) [inline CHANGELOG.md:31]:
```suggestion
- [#985](https://github.com/crypto-org-chain/cronos/pull/985) Fix versiondb verify command on older versions
```

**Review by mmsqe — APPROVED** [review]

File: integration_tests/test_versiondb.py [hunk 2]
```diff
@@ -48,7 +48,11 @@ def test_versiondb_migration(cronos: Cronos):
     print(cli1.changeset_dump(changeset_dir))
     snapshot_dir = tempfile.mkdtemp(dir=cronos.base_dir)
     print("verify and save to snapshot:", snapshot_dir)
-    _, commit_info = cli0.changeset_verify(changeset_dir, save_snapshot=snapshot_dir)
+    _, commit_info = cli0.changeset_verify(
+        changeset_dir,
+        save_snapshot=snapshot_dir,
+        target_version=1,
```
Comment (yihuang) [inline integration_tests/test_versiondb.py:54]: why does it need this?

File: versiondb/client/verify.go [hunk 3]
```diff
@@ -208,6 +202,10 @@ func verifyOneStore(tree *memiavl.Tree, store, changeSetDir, saveSnapshot string
 	}
 	// set the initial version for the store
 	initialVersion := filesWithVersion[0].Version
+	if initialVersion > uint64(targetVersion) {
```
Comment (yihuang) [inline versiondb/client/verify.go:205]:
```suggestion
	if targetVersion  > 0 && initialVersion > uint64(targetVersion) {
```

File: integration_tests/test_versiondb.py [hunk 4]
```diff
@@ -48,7 +48,11 @@ def test_versiondb_migration(cronos: Cronos):
...
+    )
```
Comment (yihuang) [inline integration_tests/test_versiondb.py:55]:
```suggestion
    _, commit_info = cli0.changeset_verify(changeset_dir, save_snapshot=snapshot_dir)
```

File: versiondb/client/verify.go [hunk 5]
```diff
@@ -208,6 +202,10 @@ func verifyOneStore(tree *memiavl.Tree, store, changeSetDir, saveSnapshot string
 	}
 	// set the initial version for the store
 	initialVersion := filesWithVersion[0].Version
+	if targetVersion  > 0 && initialVersion > uint64(targetVersion) {
```
Comment (yihuang) [inline versiondb/client/verify.go:205]:
```suggestion
	if targetVersion > 0 && initialVersion > uint64(targetVersion) {
```

**Pull request merged**